feat(COR-5931)-ALB-set-Deletion-protection-ON

dvagapov opened the pull request:
As followup task after ICM -> set for ALB attribute `Deletion protection ON` for prevent accidental 
deletion of ALB

was tested on cortest3 -> 
- added extra annotation to ing https://argocd.cortest3.us-east-2.outreach.cloud/applications/argo-rollouts?node=networking.k8s.io%2FIngress%2Fargo-rollouts%2Fargo-rollouts%2F0&resource= 
- was automatically set new attr to ALB -> https://us-east-2.console.aws.amazon.com/ec2/home?region=us-east-2#LoadBalancer:loadBalancerArn=arn:aws:elasticloadbalancing:us-east-2:182192988802:loadbalancer/app/k8s-cortest3useast2-ed70cf1a26/a4c34506f04ac5b8;tab=listeners
- Manual deletion was prevented with error `Failed to delete load balancer: Load balancer 'arn:aws:elasticloadbalancing:us-east-2:182192988802:loadbalancer/app/k8s-cortest3useast2-ed70cf1a26/a4c34506f04ac5b8' cannot be deleted because deletion protection is enabled`

zdenekvicar commented:
This looks promising, I am wondering whether we could also set this only for Production on jsonnet-lib level ?
Can I ask @kaldorn for extra eyes on this based on your ALB expertise ?

kaldorn reviewed (changes requested):
I'm not against this change for production. It's likely a good idea, and good reading of the docs for what's available. I do think it's important to communicate this change out and getting buy-in prior to turning this on. IIRC it makes it so that any ALB with this turned on will need to have it manually disabled prior to deletion -- this made initial testing with ALBs more work to iterate on so I left it disabled.

dvagapov commented:
[From off page](https://github.com/kubernetes-sigs/aws-load-balancer-controller/blob/main/docs/guide/ingress/annotations.md#custom-attributes)

I added annotation deletion_protection.enabled=true and tried to delete ing
- Can’t delete Ingress
- ALB also do not deleted
- Removed/Changed annotation to deletion_protection.enabled=false  -> ing was deleted successfully, on alb still present flag Deletion protection=On

So, I think it’s not convenient to use it in jsonnet-lib

Cancel this PR
